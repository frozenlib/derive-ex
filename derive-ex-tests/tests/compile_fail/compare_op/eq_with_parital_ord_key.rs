use derive_ex::derive_ex;

#[derive_ex(Eq, PartialEq)]
struct X(#[partial_ord(key = $.len())] String);

fn main() {}
