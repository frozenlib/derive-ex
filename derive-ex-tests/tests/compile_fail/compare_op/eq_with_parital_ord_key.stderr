error: When `#[derive_ex(Eq)]` is specified, `#[ord(key = ...)]` must be used instead of `#[partial_ord(key = ...)]`.
 --> tests/compile_fail/compare_op/eq_with_parital_ord_key.rs:4:30
  |
4 | struct X(#[partial_ord(key = $.len())] String);
  |                              ^
